talk/objectorientation/objectsclasses.tex

```diff
@@ -109,48 +142,26 @@
 
 \begin{frame}[fragile]
   \frametitlecpp[98]{Implementing methods}
-  \begin{goodpractice}{Implementing methods}
+  \begin{goodpractice}{Header/source separation}
     \begin{itemize}
-    \item usually in .cpp, outside of class declaration
-    \item using the class name as ``namespace''
-    \item short member functions can be in the header
-    \item some functions (templates, \cppinline{constexpr}) must be in the header
-    \end{itemize}
-  \end{goodpractice}
-  \begin{cppcode}
-    void MyFirstClass::squareA() {
-      a *= a;
-    }
-
-    int MyFirstClass::sum(int b) {
-      return a + b;
-    }
-  \end{cppcode}
-\end{frame}
-
-\begin{frame}[fragile]
-  \frametitlecpp[98]{\texttt{this} keyword}
-  \begin{block}{How to know an object's address?}
-    \begin{itemize}
-    \item Sometimes we need to pass a reference to ourself to a different entity
-    \item For example to implement operators, see later
-    \item All class methods can use the keyword \cppinline{this}
+    \item Functions usually implemented in .cpp, outside of class, using the class name as ``namespace''
+    \item First include header being implemented, then everything that's needed for implementation
+    \item Keep includes and code in headers to a minimum (compiled many times)
       \begin{itemize}
-        \item It returns the address of the current object
-        \item Its type is \cppinline{T*} in the methods of a class {\ttfamily T}
+        \item Short member functions can be in the header
+        \item Some functions (templates, \cppinline{constexpr}) must be in the header
```

Comment:
Previous version was less verbose and thus more readable. We should try to avoid plain sentences going on several lines and stick to one line bullets. I would also drop this change